### ISSUE TITLE
HEIC pictures not loading

### ISSUE THREAD

**Issue by dridhas** [nextcloud/maps#318]
Just installed this and the JPEG pictures with Metadata load, but not the HEIC ones.

**Comment by tacruc** [comment]
What is HEIC?

**Comment by dridhas** [comment]
Heic is the extension for the pictures taken on an iOS device.

**Comment by tacruc** [comment]
Well we're resticted on the format support un php_exiv or the library pel. Tell your iOS to use a standard format like jpg.

**Comment by mehturt** [comment]
I found this ticket about adding HEIC support to php: https://bugs.php.net/bug.php?id=77757

**Comment by dridhas** [comment]
i was able to get the HEIC pictures to show on the gallery and show thumbnails after modifying a bit imagick following this link: [https://medium.com/@eplt/5-minutes-to-install-imagemagick-with-heic-support-on-ubuntu-18-04-digitalocean-fe2d09dcef1](url).

on web browser, the metadata is not shown, but on the ios app it is.

**Comment by genebean** [comment]
bummed that this doesn't work with HEIC images. Here's hoping PHP fixes that or some other workaround is found

**Comment by tomhub** [comment]
There seems to be a way to read exif data from heic images, see: https://github.com/gino0631/nextcloud-metadata - this plugin shows heic image metadata.

**Comment by dm-vodopyanov** [comment]
+1 to @tomhub suggestion. @tacruc, could you please spend a little time to revise and re-estimate this FR? It could be easier now to implement this.

**Comment by BWibo** [comment]
+1 to this feature request. 🚀 

Today, HEIC format can be selected as default image format for the phone camera on many Android smartphones (Samsung in my case) too. I'm using it, as it saves ~40% space compared to JPG at same or even better quality.
Using `libmagickcore-6.q16-6-extra` I was able to add full support with Nextcloud (e.g. preview generation) for HEIC images. All I'm missing now is Maps support. Thx in advance.

**Comment by ignas-narbutas** [comment]
I agree. Almost all and all my new photos are in HEIC. I can view them in NC with ImageMagick. 
So I can't use this nice app until there is HEIC support.

**Comment by nemoforum** [comment]
+1 to this feature request.
It's 2023, HEIC uses less storage providing better quality.

**Comment by nemoforum** [comment]
Finally, I've found https://github.com/pulsejet/memories and it seems it covers almost everything I need.

**Comment by kylew01** [comment]
I was excited to see an update to maps today, but unfortunately there is no HEIC support still. 

Let's throw "AI" into the name, and it will be implemented tomorrow… /s

In seriousness, thank you for all of your hard work and effort.

**Comment by tacruc** [comment]
> I found this ticket about adding HEIC support to php: https://bugs.php.net/bug.php?id=77757

This is basically the issue which needs to be solved to implemt it with little work. With more work I would suggest to move to Nextcloud core metadata feature and lat that handle the metadata extraction. This is on my to-do list but not yet scheduled yet.

**Comment by hbprotoss** [comment]
> Finally, I've found https://github.com/pulsejet/memories and it seems it covers almost everything I need. Moreover, it allows h265 playback in the browser: HLS + hardware accelerated transcoding on the fly.

You saved my life
